src/pages/DatasetSearch.jsx

```diff
@@ -298,6 +298,7 @@ export const DatasetSearch = (props) => {
           });
         } catch (error) {
           Notifications.showError({ text: 'Failed to load Elasticsearch index' });
+          setLoading(false);
```

Comment:
I don't think this is desirable, since it shows an empty table, which may be misleading when on a custom data library. This means that we will not be able to tell the difference between a data table that has no data vs a data library that failed to load.

Reply:
This is true - the table doesn't know why there are no datasets, but it's also technically true that loading is complete. I noted in the PR description that I'm purposefully choosing not to try and modify the logic in the display to include an error state. If we were to go down that path, this would a larger refactor that I would prefer to do in a ticket with clear requirements in that case.

Reply:
As a user I cannot take an action if the index fails to load, that's why I don't think it's desirable to go from a loading state (where it is clear that I cannot take an action) to an empty table (where it is unclear if I can take any actions)

Reply:
> As a user I cannot take an action if the index fails to load, that's why I don't think it's desirable to go from a loading state (where it is clear that I cannot take an action) to an empty table (where it is unclear if I can take any actions)

IIUC, is having the loading spinner remain visible in the error state the intended functionality? If so, I'll be happy to close this and close out the ticket.

Reply:
Until we have a more defined approach to dealing with `index` failures (perhaps replacing the spinner with an error icon?), the spinner-as-error was the intended failure mode.